snappy/click.go

```diff
@@ -214,6 +220,47 @@ ubuntu-core-launcher {{.UdevAppName}} {{.AaProfile}} {{.Target}} "$@"
 	return templateOut.String(), nil
 }
 
+func generateSnapDesktopFile(binary Binary, pkgPath string, m *packageYaml) (string, error) {
+	desktopTemplate := `[Desktop Entry]
+Encoding=UTF-8
+Type=Application
+Name={{.Name}}
+Comment={{.Comment}}
+Icon={{.Icon}}
+Exec={{.Exec}}
```

Comment:
I realized this was an opinion, and that it was entirely possible that everyone was just putting the URL at the end of their command line and not using the flexibility of placement at all. I knew that there was a test of the click database a few weeks ago so I asked if the cache was still available, turns out it was, so I got a dump of all the Exec lines in every click package from a few weeks ago.

The vast majority of apps just put the URL at the end of the exec line. But there are some usages that I think are reasonable. Several apps have a parameter "--url=%u" where the URL gets placed in that parameter. That could probably be the last parameter, but I'm not sure how it'd handle parsing the extra space. The other usage seemed to be that people are using positional arguments, putting the URL in position 1.

While reasonable, I think that is likely these cases could be taken care of by just adding the URL as an additional parameter at the end if need be. I still believe there is a semantic reason to split binaries and applications, but the exec line formatting doesn't seem to be a problem.